crates/client/src/client/hex_formatted.rs

```diff
@@ -0,0 +1,79 @@
+use serde::{
+    de::Error,
+    Deserialize,
+    Deserializer,
+    Serialize,
+    Serializer,
+};
+use std::{
+    fmt::{
+        Debug,
+        Display,
+        Formatter,
+        LowerHex,
+    },
+    str::FromStr,
+};
+
+#[derive(Debug, thiserror::Error)]
+pub enum HexFormatError {
+    #[error("hex parsing error {0}")]
+    HexError(String),
+}
+
+#[derive(Debug, Clone, Copy, Default, PartialEq)]
+pub struct HexFormatted<T: Debug + Clone + Default + PartialEq>(pub T);
+
+impl<T: LowerHex + Debug + Clone + Default + PartialEq> Serialize for HexFormatted<T> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_str(format!("{:#x}", self.0).as_str())
```

Comment:
This formats to hex when it is serialized but the type indicates it's already hex formated. If this is intentional `pub T` should at least be `T`

Reply:
I reverted these changes now that I am not redefining any primitives. Instead, I am keep the existing `HexFormatted` struct in `schema.rs`, but that has the same questionable design. I can make your suggested change in this PR or I can look deeper into that another time.